### PR TITLE
Simulation: Build on MacOS

### LINKED ISSUE #47: CMake: gsl on MAC
Depends on #40 |

The current build uses a method of linking to libraries (such as `gsl`) that is suitable for LINUX machines, however not on MAC. 

Fortunately, `cmake` can handle this difference. The `CMakeLists.txt` in the `TissueFolding` directory should be updated in accordance with that in the `UserInterface/` directory.

### LINKED ISSUE #56: Simulation/UI: g++ specification on Mac
When building the `TissueFolding` executable on Mac, the `g++` compiler at `usr/bin/g++` is typically a linker to `Clang`, which can cause `OpenMP` not to be found, and thus the `CMake` build to fail.

The fix to this, one must install `g++` via `brew`, and then provide the path to the specific _version_ of g++ that was installed. The build readme should be updated with this.

### PR DESCRIPTION
**Pull request by willGraham01**
Closes #47 | Closes #56 |

## Build on MacOS

Adjusts the `CMakeLists.txt` file that handles the building of the simulation executable (`TissueFolding`), so that the build succeeds on MacOS.

Also adds a note regarding compiler specification on Mac to resolve #56. The `TissueFolding` `readme.md` now contains a note in its MacOS specific instructions for handling this problem. The users are advised to pass `$(which g++-xx)` where `xx` is the version number of the `g++` compiler, which will force the build to _actually_ use `g++` rather than `clang` under the alias of `g++`.

## Testing

Also adds another build test, `build_simulation_exectuable_on_mac.yml` to the CI, to confirm whether the build on MacOS is successful without PARDISO.

The new build system again suffers from the same numerical imprecision artefacts identified in #27. To combat this, [commit 500b47e](https://github.com/Yanlan-Mao-Lab/TissueOrigami/pull/48/commits/500b47e8873a347d09e85d994bbfa36abccff0b6) allows for small ( $10^{-16}$ ) numerical imprecision between the reference outputs and the generated outputs when there is no time-evolution in the system.

The build system with PARDISO has been tested, and succeeds on Ubuntu 22.04. Regression tests also pass.

The build system with PARDISO on MacOS has not been tested, since I don't have a MacOS device with a licensed PARDISO library on it. @nkhalilgharibi - if you have a MacOS with PARDISO, might be worth trying to follow the PARDISO build instructions on your Mac (in a clean clone of the repository) to see if it will still compile? The `sim_with_pardiso` tests should still run (and pass, hopefully!) once `TissueFolding` is built and moved to the right place.

### CODE REVIEW

**Comment by willGraham01** [comment]
Commenting for documentation:
- CMake appears to be finding Boost at `/usr/local/lib/cmake/Boost-1.79.0/BoostConfig.cmake` rather than `usr/local/include`.
This means that it obviously cannot find the corresponding include paths when trying to compile. Not sure why it wants to find boost here.

File: .github/workflows/build_simulation_executable_on_mac.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      #- name: tmate session
+      #  uses: mxschmitt/action-tmate@v3
+
+      # - name: Build TissueFolding with CMake
+      #   run: |
+      #     cd ${{ github.workspace }}/TissueFolding/SourceCode/
+      #     mkdir build
+      #     cd build
+      #     cmake .. -DCMAKE_CXX_COMPILER=$(which g++-11) -DCMAKE_C_COMPILER=$(which gcc-11)
+      #     cmake --build .
```
Comment (alessandrofelder) [inline .github/workflows/build_simulation_executable_on_mac.yml:34]: Can we remove this now?

**Review by alessandrofelder — APPROVED** [review]
Nice 💯 
One super-tiny suggestion.

File: .github/workflows/build_simulation_executable_on_mac.yml (hunk 1 above)
Comment (nkhalilgharibi) [reply .github/workflows/build_simulation_executable_on_mac.yml]: Trying to test this for a while now. I haven't managed to make the executable with paradise yet. Get the following error:
CMake Error at /usr/local/Cellar/cmake/3.24.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.24.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.24.2/share/cmake/Modules/FindOpenMP.cmake:545 (find_package_handle_standard_args)
  CMakeLists.txt:24 (find_package)

**Review by nkhalilgharibi — APPROVED** [review]
Thank you very much for all your work. I tested this pull request with/without pardiso on Mac:
- without pardiso: all three runs pass. With Run07008 there is warning for `growth` and `growth rate` (as before) and the CompareTxt is used. 
- with pardiso: all three runs pass. However, in addition to the same numerical difference as without pardiso, there is difference for all three runs in `Save_Frame`, `Force`, `Growth` and `TensionCompression` and CompareTxt is used.

Altogether, all looks good to me. I have added some comments in the TissueFolding/readme.md to make the instructions more clear. However, I understand that you have used all your time on the project (and even more), so I don't want this to take too much of your time. So please feel free to ignore these comments, and carry on with the merge.

**Pull request merged**